types/precison_timestamp_types.go

```diff
@@ -0,0 +1,167 @@
+package types
+
+import (
+	"fmt"
+	"github.com/cockroachdb/errors"
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+type TimePrecision int
+
+const (
+	UnknownPrecision TimePrecision = iota
+	Seconds
+	DeciSeconds
+	CentiSeconds
+	MilliSeconds
+	EMinus4Seconds // 10^-4 of seconds
+	EMinus5Seconds // 10^-5 of seconds
+	MicroSeconds
+	EMinus7Seconds // 10^-7 of seconds
+	EMinus8Seconds // 10^-8 of seconds
+	NanoSeconds
+)
+
+func timePrecisionToProtoVal(val TimePrecision) int32 {
+	switch val {
+	case Seconds:
+		return 0
+	case DeciSeconds:
+		return 1
+	case CentiSeconds:
+		return 2
+	case MilliSeconds:
+		return 3
+	case EMinus4Seconds:
+		return 4
+	case EMinus5Seconds:
+		return 5
+	case MicroSeconds:
+		return 6
+	case EMinus7Seconds:
+		return 7
+	case EMinus8Seconds:
+		return 8
+	case NanoSeconds:
+		return 9
+	}
+	panic("unreachable")
+}
+
+func ProtoToTimePrecision(val int32) (TimePrecision, error) {
+	switch val {
+	case 0:
+		return Seconds, nil
+	case 1:
+		return DeciSeconds, nil
+	case 2:
+		return CentiSeconds, nil
+	case 3:
+		return MilliSeconds, nil
+	case 4:
+		return EMinus4Seconds, nil
+	case 5:
+		return EMinus5Seconds, nil
+	case 6:
+		return MicroSeconds, nil
+	case 7:
+		return EMinus7Seconds, nil
+	case 8:
+		return EMinus8Seconds, nil
+	case 9:
+		return NanoSeconds, nil
+	}
+	return UnknownPrecision, errors.Newf("invalid TimePrecision value %d", val)
```

Comment:
```suggestion
        return UnknownPrecision, fmt.Errorf("invalid TimePrecision value: %d", val)
```

Reply:
Replied in [this](https://github.com/substrait-io/substrait-go/pull/41#discussion_r1710861808) comment
Since currently, reverted to use fmt.Errorf so resolving this comment